[READY] Fix GoTo lists arriving at the wrong column.

puremourning opened the pull request:
## Problem

When a `GoTo` command returns a list of destinations, such as declarations and definitions in Python and references in JavaScript, we pop the Vim quick fix list with a list of destinations. however, we always jump to the character before the actual declaration/definition.

## Resolution

Vim's QuickFix lists require 1-based columns, which is what is returned from ycmd's commands, so don't subtract 1 from the column number.

As noted in the comments, the Vim documentation for `setqflist` is somewhat vague about this "byte offset", but it is confirmed to mean "1-based column number" both in testing and in `:help getqflist`.

## Manual testing

Tested this with the Tern completer (`:YcmCompleter GoToReferences`) and Jedi completer (`:YcmCompleter GoToDeclaration`). For the Jedi completer, use something like:

```python
x = 1
if False:
  x = 2
else:
  x = 3

print( str(x) )
```

Then `:YcmCompleter GoToDeclaration` on `x`. This returns a list, which pops up the quick fix window.

*Previous behaviour*

Previously, the jump would go to the character before the actual declaration (i.e. the `<space>` before `x` in each line)

## Automated testing

Just some simple execution of the changed code, proving that we don't take 1 from the column.

<!-- Reviewable:start -->
[<img src="https://reviewable.io/review_button.png" height=40 alt="Review on Reviewable"/>](https://reviewable.io/reviews/valloric/youcompleteme/1876)
<!-- Reviewable:end -->

puremourning commented:
The YCM client AppVeyor tests are failing with an obscure error:

```

python run_tests.py
-- The C compiler identification is unknown
-- The CXX compiler identification is unknown
CMake Error at CMakeLists.txt:39 (project):
  No CMAKE_C_COMPILER could be found.
 
 
 
CMake Error at CMakeLists.txt:39 (project):
  No CMAKE_CXX_COMPILER could be found.
 
 ```

I don’t think this has anything to do with my change :/

micbou commented:
Yes, I am looking into it.

---

Review status: 0 of 3 files reviewed at latest revision, 3 unresolved discussions, some commit checks failed.

---

*[python/ycm/client/command_request.py, line 142 \[r1\]](https://reviewable.io:443/reviews/valloric/youcompleteme/1876#-K78jnK8sqtwrayeg2AR:-K78jnK8sqtwrayeg2AS:-520954389) ([raw file](https://github.com/valloric/youcompleteme/blob/fe9f68e66b5a5d0eeef8b1539d7558852ba76efb/python/ycm/client/command_request.py#L142)):*
`#` leftover.

---

*[python/ycm/client/tests/command_request_test.py, line 2 \[r1\]](https://reviewable.io:443/reviews/valloric/youcompleteme/1876#-K78k-YzaUxjNCMC4lg6:-K78k-Z-MRFYhW2FmZuv:1734452677) ([raw file](https://github.com/valloric/youcompleteme/blob/fe9f68e66b5a5d0eeef8b1539d7558852ba76efb/python/ycm/client/tests/command_request_test.py#L2)):*
I heard that we are in 2016 now.

---

*[python/ycm/vimsupport.py, line 248 \[r1\]](https://reviewable.io:443/reviews/valloric/youcompleteme/1876#-K78lgVGg1oTFVTbeZ92:-K78lgVH_BUjhUgW0VoM:1696047641) ([raw file](https://github.com/valloric/youcompleteme/blob/fe9f68e66b5a5d0eeef8b1539d7558852ba76efb/python/ycm/vimsupport.py#L248)):*
Last quote and parenthesis are not in the right order.

---


*Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/valloric/youcompleteme/1876#-:-K78jj3xD0CUWs0f-lmO:1837193607)*
<!-- Sent from Reviewable.io -->

puremourning commented:
Review status: 0 of 3 files reviewed at latest revision, 3 unresolved discussions.

---

*[python/ycm/client/command_request.py, line 142 \[r1\]](https://reviewable.io:443/reviews/valloric/youcompleteme/1876#-K78jnK8sqtwrayeg2AR:-K78nnjMhk7FnwxcbZ8E:-467915377) ([raw file](https://github.com/valloric/youcompleteme/blob/fe9f68e66b5a5d0eeef8b1539d7558852ba76efb/python/ycm/client/command_request.py#L142)):*
fixed

---

*[python/ycm/client/tests/command_request_test.py, line 2 \[r1\]](https://reviewable.io:443/reviews/valloric/youcompleteme/1876#-K78k-YzaUxjNCMC4lg6:-K78noqba2b4-amRDabc:-786650187) ([raw file](https://github.com/valloric/youcompleteme/blob/fe9f68e66b5a5d0eeef8b1539d7558852ba76efb/python/ycm/client/tests/command_request_test.py#L2)):*
Yikes. When did that happen?

---

*[python/ycm/vimsupport.py, line 248 \[r1\]](https://reviewable.io:443/reviews/valloric/youcompleteme/1876#-K78lgVGg1oTFVTbeZ92:-K78nxG10cMIrQtqgrcM:-497468049) ([raw file](https://github.com/valloric/youcompleteme/blob/fe9f68e66b5a5d0eeef8b1539d7558852ba76efb/python/ycm/vimsupport.py#L248)):*
Fixed

---


*Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/valloric/youcompleteme/1876)*
<!-- Sent from Reviewable.io -->

micbou commented:
For now, I can only say that this is an issue with MSVC 14.

---

Reviewed 1 of 3 files at r2.
Review status: 1 of 3 files reviewed at latest revision, all discussions resolved, some commit checks failed.

---



*Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/valloric/youcompleteme/1876#-:-K799K-FxWN6pNsfj0Y9:-2097634985)*
<!-- Sent from Reviewable.io -->

micbou commented:
This should be fixed but I don't have the permissions to restart the build. @puremourning You can amend your branch without making any changes and push force it or we can wait for @Valloric.

---

Review status: 1 of 3 files reviewed at latest revision, all discussions resolved, some commit checks failed.

---



*Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/valloric/youcompleteme/1876#-:-K7Cagb2PBKXzzJUZLMB:-1659260551)*
<!-- Sent from Reviewable.io -->

puremourning commented:
Thanks @micbou I have pushed. Let's see if it works :pray: 

---

Review status: 1 of 3 files reviewed at latest revision, all discussions resolved.

---



*Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/valloric/youcompleteme/1876#-:-K7DNGayiIa2W0-zQjfc:1799599529)*
<!-- Sent from Reviewable.io -->

puremourning commented:
Cool, this is now ready (Appveyor fixed)

vheon commented:
Tests are passing, simple change backed up by tests and explaining comments... what to say other then <img class="emoji" title=":lgtm:" alt=":lgtm:" align="absmiddle" src="https://reviewable.io/lgtm.png" height="20" width="61"/> ? :+1: 

---

Reviewed 3 of 3 files at r2.
Review status: all files reviewed at latest revision, all discussions resolved.

---



*Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/valloric/youcompleteme/1876#-:-K7Dp9_n-bQhgAOG3mCr:2094214714)*
<!-- Sent from Reviewable.io -->

micbou commented:
Reviewed 2 of 3 files at r2.
Review status: all files reviewed at latest revision, 1 unresolved discussion, all commit checks successful.

---

*[python/ycm/client/tests/command_request_test.py, line 33 \[r3\]](https://reviewable.io:443/reviews/valloric/youcompleteme/1876#-K7Df3U96Eq0EUxh31oP:-K7Df3U96Eq0EUxh31oQ:110441627) ([raw file](https://github.com/valloric/youcompleteme/blob/7f7b2dac2835c1a42ac331ed3a45ed43b4c98001/python/ycm/client/tests/command_request_test.py#L33)):*
Space and quote inversed at the end of `GoToTest`.

---


*Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/valloric/youcompleteme/1876)*
<!-- Sent from Reviewable.io -->

puremourning commented:
Review status: all files reviewed at latest revision, 1 unresolved discussion.

---

*[python/ycm/client/tests/command_request_test.py, line 33 \[r3\]](https://reviewable.io:443/reviews/valloric/youcompleteme/1876#-K7Df3U96Eq0EUxh31oP:-K7DscTAZ72rZjYkP0uV:-302373310) ([raw file](https://github.com/valloric/youcompleteme/blob/7f7b2dac2835c1a42ac331ed3a45ed43b4c98001/python/ycm/client/tests/command_request_test.py#L33)):*
Good spot. Fixed. It doesn't affect the behaviour of the test.

---


*Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/valloric/youcompleteme/1876)*
<!-- Sent from Reviewable.io -->

micbou commented:
<img class="emoji" title=":lgtm:" alt=":lgtm:" align="absmiddle" src="https://reviewable.io/lgtm.png" height="20" width="61"/>

@homu r+

---

Reviewed 1 of 1 files at r4.
Review status: all files reviewed at latest revision, all discussions resolved.

---



*Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/valloric/youcompleteme/1876#-:-K7DwZWcZ5FHl5zQUqwo:187772251)*
<!-- Sent from Reviewable.io -->

homu commented:
:pushpin: Commit 134b06b has been approved by `micbou`

<!-- @homu r=micbou 134b06b65e5f66842cbf5543d0ad1edf45053f24 -->

homu commented:
:zap: Test exempted - [status](https://travis-ci.org/Valloric/YouCompleteMe/builds/100236104)

Valloric commented:
Awesome stuff! 

---

Review status: all files reviewed at latest revision, 1 unresolved discussion, all commit checks successful.

---

*[python/ycm/client/tests/command_request_test.py, line 91 \[r4\]](https://reviewable.io:443/reviews/valloric/youcompleteme/1876#-K7EqBaBLBrTA_hj41pK:-K7EqBaBLBrTA_hj41pL:2095508616) ([raw file](https://github.com/valloric/youcompleteme/blob/134b06b65e5f66842cbf5543d0ad1edf45053f24/python/ycm/client/tests/command_request_test.py#L91)):*
LORD TESTIVUS IS VERY PLEASED WITH YOUR OFFERING.

:D

---


*Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/valloric/youcompleteme/1876#-:-K7EqS_iS9oXzOGQ1eGf:1508746897)*
<!-- Sent from Reviewable.io -->
